### PR TITLE
Adds diagrams to the multi-key how it works page

### PR DESCRIPTION
**Pull request by GBKS**
For issue #561. This adds diagrams for the multi-key page to explain more visually how a wallet is constructed and transactions signed, without going into too much technical detail.

If we like this type of diagram, I can also create variations for the other key management pages.

🛰️[Check the preview](https://deploy-preview-1064--bitcoin-design-site.netlify.app/guide/how-it-works/private-key-management/multi-key/#general-user-flow)🛸

### CODE REVIEW

File: assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile.png
Comment (yashrajd) [inline assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile.png:1]: Suggestions:
- the arrows taken together could form a circle so the visual looks better
- the key icons themselves could be shown on/near each arrow if it does not clutter the visual

Comment (yashrajd) [reply assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile.png:1]: Any of the above-mentioned changes would need to be applied to all versions (sizes) of this image.

File: guide/how-it-works/private-key-management/multi-key.md [hunk 1]
```diff
@@ -74,7 +76,57 @@ A software wallet application or coordination software initiates a multi-sig wal
...
+   image = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange.png"
+   retina = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange@2x.png"
+   mobile = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile.png"
+   mobileRetina = "/assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile@2x.png"
+   alt-text = "Diagram of the three wallet owners exchanging public keys"
+   width = 800
+   height = 381
+%}
+
+They can now individually construct the same multi-key wallet with the 3 public keys and the spending conditions they decided on (2-of-3). Addresses generated by each wallet will be identical. The order of the public keys is important here. A different order will result in a different wallet.
```
Comment (yashrajd) [inline guide/how-it-works/private-key-management/multi-key.md:107]: The order of the public keys can be ensured by the software, so it is not necessary for the signers themselves to sign in any particular order. Unsure if this necessitates any change to the text here.

File: guide/how-it-works/private-key-management/multi-key.md [hunk 2]
```diff
@@ -74,7 +76,57 @@ A software wallet application or coordination software initiates a multi-sig wal
 
 {% include fact/close.html %}
 
-### Best practice
+### General user flow
+
+Alice, Robert, and Charles want to set up a multi-key wallet that has a total of 3 keys, with 2 keys required to send a transaction.
+
+In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for initiating and approving transactions, and will never share it it. The public key is used for construct the multi-key wallet for generating addresses to receive bitcoin to.
```
Comment (yashrajd) [inline guide/how-it-works/private-key-management/multi-key.md:83]: Don't think the private key is required for "initiating" a transaction, only for signing/approving. Perhaps I misunderstand what you mean here...

Comment (yashrajd) [inline guide/how-it-works/private-key-management/multi-key.md:83]:
```suggestion
In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for approving transactions, and will never share it it. The public key is shared with the other signers, and all the public keys are together used to generate addresses to receive bitcoin to.
```

**Comment by yashrajd** [comment]
I just realised this PR was only for the diagrams, and most of my inputs were about the text.

File: guide/how-it-works/private-key-management/multi-key.md (hunk 2 above)
Comment (rabbitholiness) [inline guide/how-it-works/private-key-management/multi-key.md:83]:
```suggestion
In the first step, each of them independently generate a private key and a public key (technically an extended public key). They will use the private key for initiating and approving transactions, and will never share it it. The public key is used to construct the multi-key wallet and for generating addresses to receive bitcoin to.
```

File: guide/how-it-works/private-key-management/multi-key.md (hunk 1 above)
Comment (rabbitholiness) [reply guide/how-it-works/private-key-management/multi-key.md:107]: I think this would be a good addition.

**Comment by rabbitholiness** [comment]
The diagrams look great. Friendly and easy to understand.

File: guide/how-it-works/private-key-management/multi-key.md [hunk 3]
```diff
@@ -97,13 +149,13 @@ A software wallet application or coordination software initiates a multi-sig wal
...
+- [Sparrow Wallet](https://sparrowwallet.com/)
 - [Casa](https://keys.casa) co-managed 2-of-3, or 3-of-5
 - [Electrum](https://electrum.org)
-- [Bluewallet](https://bluewallet.io)
+- [BlueWallet](https://bluewallet.io)
 - [Unchained Capital/Caravan](https://unchained-capital.com) co-managed
 - [Specter](https://specter.solutions)
 - [Armory](https://btcarmory.com)
-- [Guarda](https://guarda.com)
 - [Revault](https://revault.dev){:target="_blank"} - in development
```
Comment (rabbitholiness) [inline guide/how-it-works/private-key-management/multi-key.md:160]: Maybe Nunchuk would also be a nice addition?

File: assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile.png
Comment (GBKS) [reply assets/images/guide/how-it-works/private-key-management/multi-key/multi-key-diagram-1-key-exchange-mobile.png:1]: I had intentionally kept them separate to indicate that there is a mutual exchange between each of the parties.

File: guide/how-it-works/private-key-management/multi-key.md (hunk 2 above)
Comment (GBKS) [reply guide/how-it-works/private-key-management/multi-key.md:83]: I don't see where it says that in the text...?

File: guide/how-it-works/private-key-management/multi-key.md (hunk 1 above)
Comment (GBKS) [reply guide/how-it-works/private-key-management/multi-key.md:107]: The order has to be the same for constructing the wallet, but not for signing. I tested this with the [BitcoinJS library](https://github.com/bitcoinjs/bitcoinjs-lib) earlier this week.

In the wallet coordination file, there are fingerprints of each key. That's how the order can be tracked later once the wallet is set up.

File: guide/how-it-works/private-key-management/multi-key.md (hunk 3 above)
Comment (GBKS) [reply guide/how-it-works/private-key-management/multi-key.md:161]: Added.

File: guide/how-it-works/private-key-management/multi-key.md (hunk 1 above)
Comment (GBKS) [reply guide/how-it-works/private-key-management/multi-key.md]: I added an extra note about this.

**Comment by GBKS** [comment]
Thanks for the feedback. Since both of you had notes on the paragraph at line 83, I combined the changes and did not replay to your individual comments. Hope it reads better now. And ready for another round of review.

File: guide/how-it-works/private-key-management/multi-key.md [hunk 4]
```diff
@@ -40,15 +40,15 @@ https://www.figma.com/community/file/995256542920917246/BDG---Private-key-manage
 
 All of the previous schemes have relied on a single private key to control the wallet. This presents an all-or-nothing risk for loss of funds from both theft and negligence. To counter this, a wallet can have several private keys attached of which all or a subset need to sign any transactions.
 
-This is often called multi-signature, or multisig for short, but is also sometimes referred to as a *vault*. A multi-key setup is described as *n-of-n* to indicate how many keys are needed to sign a transaction out of the issued number. For example, a *2-of-3* setup requires two of the three private keys to sign a transaction for it to be valid.
+This is often called multi-signature, or multi-sig for short, but is also sometimes referred to as a *vault*. A multi-key setup is described as *n-of-n* to indicate how many keys are needed to sign a transaction out of the issued number. For example, a *2-of-3* setup requires two of the three private keys to sign a transaction for it to be valid.
```
Comment (yashrajd) [inline guide/how-it-works/private-key-management/multi-key.md:43]:
```suggestion
This is often called multi-signature, or multi-sig for short, but is also sometimes referred to as a *vault*. A multi-key setup is described as *m-of-n* to indicate how many keys are needed to sign a transaction out of the issued number. For example, a *2-of-3* setup requires two of the three private keys to sign a transaction for it to be valid.
```

File: guide/how-it-works/private-key-management/multi-key.md [hunk 5]
```diff
@@ -74,7 +76,57 @@ A software wallet application or coordination software initiates a multi-sig wal
 
 {% include fact/close.html %}
 
-### Best practice
+### General user flow
+
+Alice, Robert, and Charles want to set up a multi-key wallet that has a total of 3 keys, with 2 keys required to send a transaction.
+
+In the first step, each of them independently generates a private key and a public key (technically an extended public key). All 3 public keys together are used to construct the multi-key wallet, for generating addresses to receive bitcoin to, and in extension also for constructing new transactions. The private key is never shared, and used to approve transactions, initiated by themselves or others.
```
Comment (yashrajd) [inline guide/how-it-works/private-key-management/multi-key.md:83]: nit

```suggestion
In the first step, each of them independently generates a private key and a public key (technically an extended public key). All 3 public keys together are used to construct the multi-key wallet, for generating addresses to receive bitcoin to, and by extension also for constructing new transactions. The private key is never shared, and used to approve transactions, initiated by themselves or others.
```

**Review by yashrajd — CHANGES_REQUESTED** [review]
The images are a great addition, and look great too.

Added a final couple items in feedback.

**Comment by GBKS** [comment]
Thanks for the feedback, I just accepted both changes.

I wonder how consistent we are across the guide with terms like multi-sig vs multisig, blockchain vs block chain, etc. [Bitcoin OpTech](https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md#compound-words) has a style guide that I refer to when in doubt. Not super urgent, but maybe something to revise at some point.

**Comment by mouxdesign** [comment]
Those diagrams are great. Super help for understanding the concept of exchanging keys. 
+1 for creating variations on other key management practices.

**Review by mouxdesign — APPROVED** [review]

**Pull request merged**